packages/template-retail-react-app/app/pages/account/order-history.jsx

```diff
@@ -22,30 +22,85 @@ import {
 import useNavigation from '../../hooks/use-navigation'
 import {usePageUrls, useSearchParams} from '../../hooks'
-import {useAccountOrders} from './util/order-context'
 import PageActionPlaceHolder from '../../components/page-action-placeholder'
 import Link from '../../components/link'
 import {ChevronRightIcon, ReceiptIcon} from '../../components/icons'
 import Pagination from '../../components/pagination'
+import PropTypes from 'prop-types'
+
+const DEFAULT_PAGINATION_LIMIT = 10
+const OrderProductImages = ({productItems}) => {
+    const ids = productItems.map((item) => item.productId).join(',') ?? ''
+    const {data: {data: products} = {}, isLoading} = useProducts({
+        parameters: {
+            ids: ids
+        }
+    })
+
+    const images = products?.map((product) => {
+        return product?.imageGroups?.find((group) => group.viewType === 'small').images[0]
+    })
+
+    return (
+        <>
+            {!isLoading && products
+                ? images.map((image, index) => {
+                      return (
+                          <AspectRatio
+                              key={index}
+                              ratio={1}
+                              width="88px"
+                              w="88px"
+                              borderRadius="base"
+                              overflow="hidden"
+                          >
+                              <Img
+                                  alt={image?.alt}
+                                  src={image?.disBaseLink || image?.link}
+                                  fallback={<Box background="gray.100" boxSize="full" />}
+                              />
+                          </AspectRatio>
+                      )
+                  })
+                : productItems.map((item, index) => {
+                      return <Skeleton key={index} h="88px" w="88px" />
+                  })}
+        </>
+    )
+}
+OrderProductImages.propTypes = {
+    productItems: PropTypes.array
+}
 
 const AccountOrderHistory = () => {
     const location = useLocation()
     const {formatMessage, formatDate} = useIntl()
-    const searchParams = useSearchParams({limit: 10, offset: 0})
     const navigate = useNavigation()
-    const customer = useCustomer()
-    const {orderIdsByOffset, ordersById, productsById, isLoading, fetchOrders, paging} =
-        useAccountOrders()
-    const pageUrls = usePageUrls({total: paging.total, limit: paging.limit})
 
-    const orders =
-        orderIdsByOffset[searchParams.offset || 0]?.map((orderId) => ordersById[orderId]) || []
+    const {data: customer, isLoading: isLoadingCustomer} = useCurrentCustomer()
+    const {customerId} = customer
+
+    const searchParams = useSearchParams()
+    const {limit, offset} = searchParams[0]
+
+    const {data: {data: orders, ...paging} = {}, isLoading: isLoadingOrders} = useCustomerOrders(
+        {
+            parameters: {customerId, limit: DEFAULT_PAGINATION_LIMIT || limit, offset}
```

Comment:
`DEFAULT_PAGINATION_LIMIT || limit` feels weird, since it's a constant (and probably not `0`).

Reply:
I believe we will use the constant if limit is not defined in the searchParam. I don't find it weird though 🤔

Reply:
The `DEFAULT_PAGINATION_LIMIT` was introduced mainly for the PR review.

The `limit` vale from the API is always 25 and I didn't want to create 25 orders to test the changes.